src/core/maps/grid_cell_factory.h

```diff
@@ -15,6 +26,10 @@ class GridCell {
   virtual double obst_y() const { return 0; }
 };
 
+/**
+ * \brief This is abstract class, whose subclasses are responsible for strategy of cell's creation.
```

Comment:
Slightly better variant: "The base class for factories that incapsulate creation of a specific cell (Factory method pattern is applied)."